test/runit/tests/heatmap_summary.R

```diff
@@ -13,6 +13,49 @@ test_set2_measurements <- test_set2[ , c("GSM210004", "GSM210005", "GSM210006",
 test_data <- list("n0_s1" = test_set)
 test_data_measurements <- list("n0_s1" = test_set_measurements)
 
+test_set_preprocessed  <- test_set
+colnames(test_set_preprocessed) <- c("Row.Label" , "Bio.marker","GSM210004_n0_s1", "GSM210005_n0_s1", "GSM210006_n1_s1", "GSM210007_n1_s1")
+test_set_preprocessed <- cbind(test_set_preprocessed, test_set2_measurements, stringsAsFactors = F)
+colnames(test_set_preprocessed) <- c("Row.Label" , "Bio.marker","GSM210004_n0_s1", "GSM210005_n0_s1", "GSM210006_n1_s1", "GSM210007_n1_s1",
+                                     "GSM210004_n0_s2", "GSM210005_n0_s2", "GSM210006_n1_s2", "GSM210007_n1_s2")
+### unit tests for function get_input_data ###
+assign("loaded_variables", test_data, envir = .GlobalEnv)
+assign("preprocessed", test_set_preprocessed, envir = .GlobalEnv)
+
+## get_input_data should return value of loaded_variables if phase = "fetch"
+test.get_input_data.fetch <- function(){
+  fetched_data <- get_input_data("fetch")
+  checkEquals(loaded_variables, fetched_data)
+}
+
+## get_input_data should take variable preprocessed and return its contents as a named list if phase = "preprocess"
+test.get_input_data.preprocess <- function(){
+  checkEquals(list("preprocessed" = preprocessed), get_input_data("preprocess"))
+}
+
+## get_input_data should throw error if phase is NA or anything other than "fetch" and "preprocess
+test.get_input_data.phaseNA <- function(){
+  checkException(get_input_data(NA))
+}
+
+test.get_input_data.phaseIncorrect <- function(){
+  checkException(get_input_data("Incorrect_phase"))
+}
+
+## get_input_data should throw error if phase is fetch and loaded_variables does not exist
+test.get_input_data.no_loaded_variables <- function(){
+  rm(loaded_variables, envir = .GlobalEnv)
+  checkException(get_input_data("fetch"))
+  assign("loaded_variables", test_data, envir = .GlobalEnv)
```

Comment:
It would be better to put cleanUp and setupCode in 
 
    .setUp <- function() {
      assign("loaded_variables", test_data, envir = .GlobalEnv)
      assign("preprocessed", test_set_preprocessed, envir = .GlobalEnv)
    }

    .tearDown <- function() {
     # clean up code
    }

It is less error prone.  In this test if last line does not get executed (or you remove it by accident) all the remaining tests will fail.